core/unit_test/TestReduce.hpp

```diff
@@ -625,4 +625,22 @@ TEST(TEST_CATEGORY, int_combined_reduce_mixed) {
 }
 #endif
 #endif
+
+#if defined(KOKKOS_ENABLE_CUDA)
```

Comment:
Why restrict the test to just `Cuda`? How are the other backends behaving?

Reply:
agreed we should check this always.

Reply:
I will update the PR

Reply:
@masterleinad sorry i saw your comment but forgot to answer: the reason i kept this guarded is that originally i wanted to proceed one backend at a time, so i first wanted to ensure the cuda case worked since it was reported in the issue and get that merged. Then i was going to try this on the other backends . 

@crtrott i am updating it now to run alwasy